### PR TITLE
test(js): Convert `issueList/overview` test to RTL

### PR DESCRIPTION
**Pull request by billyvg**

### CODE REVIEW

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 1]
```diff
@@ -1112,696 +1167,11 @@ describe('IssueList', function () {
...
-        groupIds: ['1'],
-        loading: false,
-      });
-
-      const issues = wrapper.find('ProcessingIssueList');
-      expect(issues).toHaveLength(1);
-    });
-  });
-
-  describe('render states', function () {
```
Comment (scttcper) [inline tests/js/spec/views/issueList/overview.spec.jsx:1365]: todo

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 2]
```diff
@@ -1112,696 +1167,11 @@ describe('IssueList', function () {
...
-        datetime: {period: '24h'},
-      };
-      wrapper.setProps({selection});
-      wrapper.update();
-
-      expect(fetchTagsRequest).toHaveBeenCalledTimes(2);
-    });
-  });
-
-  describe('processingIssues', function () {
```
Comment (scttcper) [inline tests/js/spec/views/issueList/overview.spec.jsx:1344]: todo?

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 3]
```diff
@@ -1112,696 +1167,11 @@ describe('IssueList', function () {
...
-      expect(members[project.slug]).toBeTruthy();
-      expect(members[project.slug][0].email).toBeTruthy();
-    });
-
-    it('fetches groups when there is no searchid', async function () {
-      await wrapper.instance().componentDidMount();
-    });
-  });
-
-  describe('componentDidUpdate fetching groups', function () {
```
Comment (scttcper) [inline tests/js/spec/views/issueList/overview.spec.jsx:1239]: todo

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 4]
```diff
@@ -1112,696 +1167,11 @@ describe('IssueList', function () {
...
-        query: {
-          environment: [],
-          project: props.selection.projects,
-          statsPeriod: '14d',
-        },
-      });
-    });
-  });
-
-  describe('getEndpointParams', function () {
```
Comment (scttcper) [inline tests/js/spec/views/issueList/overview.spec.jsx:1162]: todo

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 5]
```diff
@@ -1112,696 +1167,11 @@ describe('IssueList', function () {
...
-
-    it('uses saved search data', function () {
-      const value = wrapper.instance().getEndpointParams();
-
-      expect(value.query).toEqual('is:unresolved');
-      expect(value.project).toEqual([parseInt(savedSearch.projectId, 10)]);
-    });
-  });
-
-  describe('componentDidMount', function () {
```
Comment (scttcper) [inline tests/js/spec/views/issueList/overview.spec.jsx:1207]: todo

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 6]
```diff
@@ -126,35 +120,16 @@ describe('IssueList', function () {
...
   });
 
   afterEach(function () {
+    cleanup();
     jest.clearAllMocks();
     MockApiClient.clearMockResponses();
-    if (wrapper) {
-      wrapper.unmount();
+
+    for (const el of document.querySelectorAll('style').values()) {
```
Comment (scttcper) [inline tests/js/spec/views/issueList/overview.spec.jsx:130]: not sure if we need this @billyvg

File: package.json [hunk 7]
```diff
@@ -187,6 +187,7 @@
   "resolutions": {
     "**/rrweb": "^0.7.33",
     "**/is-svg": ">=4.2.2",
+    "aria-query": "https://github.com/getsentry/aria-query#0b25434ec87e77f7136aa36d06d155954cb872a4",
```
Comment (scttcper) [inline package.json:190]: can probably pull this change into another pr

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 6 above)
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx:130]: Ah yeah I have some perf related experiments in here. These don't seem to amount to much.

File: static/app/views/issueList/overview.tsx [hunk 8]
```diff
@@ -1129,7 +1118,6 @@ class IssueListOverview extends React.Component<Props, State> {
                 selection={selection}
                 query={query}
                 queryCount={modifiedQueryCount}
-                displayCount={displayCount}
```
Comment (billyvg) [inline static/app/views/issueList/overview.tsx:1132]: Was this not used?

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 9]
```diff
@@ -674,358 +603,652 @@ describe('IssueList', function () {
...
+          page: 1,
+          environment: [],
+          project: [],
+          query: 'is:unresolved',
+          statsPeriod: '14d',
+        },
+      });
     });
 
-    it('pinning and unpinning searches should keep project selected', async function () {
```
Comment (billyvg) [inline tests/js/spec/views/issueList/overview.spec.jsx:807]: Was this test removed?

File: static/app/views/issueList/overview.tsx (hunk 8 above)
Comment (scttcper) [reply static/app/views/issueList/overview.tsx:1132]: nope

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 9 above)
Comment (scttcper) [reply tests/js/spec/views/issueList/overview.spec.jsx:807]: i've combined this one with `pins and unpins a saved query` it just checks the project id now too

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 10]
```diff
@@ -175,10 +146,12 @@ describe('IssueList', function () {
 
     /* helpers */
     const getSavedSearchTitle = w =>
-      w.find('SavedSearchTab DropdownMenu a').text().trim();
+      w.queryByTestId('saved-search-title')?.textContent.trim();
 
     const getSearchBarValue = w =>
-      w.find('SmartSearchBarContainer textarea').prop('value').trim();
+      w
+        .queryByPlaceholderText('Search for events, users, tags, and more')
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:153]: should we use `findByPlaceholderText` instead of `queryByPlaceholderText` ?

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 11]
```diff
@@ -674,358 +603,652 @@ describe('IssueList', function () {
...
+            projects,
+          }),
+        });
+
+        await waitFor(() => {
+          expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+        });
+
+        await waitFor(() =>
+          expect(wrapper.getByTestId('awaiting-events')).toBeInTheDocument()
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:1100]: should we check for a rendered text instead a `data-test-id?` for instance `Waiting for events…`

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 12]
```diff
@@ -674,358 +603,652 @@ describe('IssueList', function () {
...
-    it.todo('saves a new query');
+    it('fetches and displays processing issues', async function () {
+      createWrapper();
+
+      await waitFor(() => {
+        expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+      });
 
-    it.todo('loads pinned search when invalid saved search id is accessed');
+      expect(wrapper.getByTestId('processing-issue-hint')).toHaveTextContent(
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:842]: should we `queryByText` here?

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 13]
```diff
@@ -674,358 +603,652 @@ describe('IssueList', function () {
...
-      expect(browserHistory.push).toHaveBeenLastCalledWith(
-        expect.objectContaining({
-          pathname: '/organizations/org-slug/issues/searches/666/',
-          query: expect.objectContaining({
-            project: [123],
-            environment: ['prod'],
-            query: 'assigned:me level:fatal',
-          }),
-        })
+      expect(wrapper.getByTestId('issues-pagination')).toHaveTextContent(
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:763]: should we use `getByText` here?

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 14]
```diff
@@ -465,51 +419,54 @@ describe('IssueList', function () {
...
       });
       createWrapper();
-      await tick();
-      await tick();
-      wrapper.update();
 
-      wrapper.find('SavedSearchTab DropdownMenu a').simulate('click');
-      wrapper.find('SavedSearchMenuItem a').last().simulate('click');
+      await waitFor(() => {
+        expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
```
[the remark this answers is not in the document]
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx:433]: Why `findBy` vs `queryBy`?

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 11 above)
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx:1100]: Why?

My assumption here is that querying by test id is more efficient than querying by text, but I haven't confirmed this is true.

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 15]
```diff
@@ -24,19 +28,16 @@ const DEFAULT_LINKS_HEADER =
   '<http://127.0.0.1:8000/api/0/organizations/org-slug/issues/?cursor=1443575000:0:0>; rel="next"; results="true"; cursor="1443575000:0:0"';
 
 describe('IssueList', function () {
+  /**  @type {ReturnType<typeof mountWithTheme>} */
```
[the remark this answers is not in the document]
Comment (scttcper) [reply tests/js/spec/views/issueList/overview.spec.jsx:31]: the autocomplete this gives was nice. otherwise wrapper type is `any`

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply tests/js/spec/views/issueList/overview.spec.jsx:433]: because `findBy querie`s can be used to await the changes in the DOM

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 11 above)
Comment (priscilawebdev) [reply tests/js/spec/views/issueList/overview.spec.jsx:1100]: hm wouldn't it be better to test what information users see on the screen?

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply tests/js/spec/views/issueList/overview.spec.jsx:433]: https://testing-library.com/docs/dom-testing-library/api-async#findby-queries

[the remark this answers is not in the document]
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx:433]: Ahh that makes sense

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 10 above)
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx]: We were using this like

```javascript
await waitFor(() => expect(getSearchBarValue(wrapper)).toBe('is:unresolved'));
```

I've updated it to be this instead

```javascript
await wrapper.findByDisplayValue('is:unresolved');
```

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 16]
```diff
@@ -175,10 +146,12 @@ describe('IssueList', function () {
 
     /* helpers */
     const getSavedSearchTitle = w =>
-      w.find('SavedSearchTab DropdownMenu a').text().trim();
+      w.queryByTestId('saved-search-title')?.textContent.trim();
```
[the remark this answers is not in the document]
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx]: We were using this like
```javascript
expect(getSavedSearchTitle(wrapper)).toBe('Saved Searches');
```

I've updated it to be

```javascript
      const savedSearches = await wrapper.findAllByText('Saved Searches');
      expect(savedSearches).toHaveLength(1);
      expect(savedSearches[0]).toBeInTheDocument();
```

This method of testing seems odd to be, but seems to be the common practice for RTL, e.g. what happens if something else in the screen has `Saved Searches`?

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx:440]: So actually, I think we have to use `waitFor` here because we're waiting for the element to *not* exist, I don't think this can be done with `findBy` queries.

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 13 above)
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx]: I've changed these, but they are kind of a pain because there are multiple `<span/>`s that wrap text and there's no native RTL way to use `findByText`, I had to use a helper function that I found from stackoverflow.

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 11 above)
Comment (priscilawebdev) [reply tests/js/spec/views/issueList/overview.spec.jsx:1087]: > In the spirit of the guiding principles, it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post "Making your UI tests resilient to change"

https://testing-library.com/docs/queries/bytestid/

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 17]
```diff
@@ -1,16 +1,25 @@
...
 
 import StreamGroup from 'app/components/stream/group';
-import GroupStore from 'app/stores/groupStore';
 import TagStore from 'app/stores/tagStore';
-import * as parseLinkHeader from 'app/utils/parseLinkHeader';
+import IssueListActions from 'app/views/issueList/actions';
 import IssueListWithStores, {IssueListOverview} from 'app/views/issueList/overview';
 
+// Taken from https://stackoverflow.com/a/56859650/1015027
+const withMarkup = query => text =>
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:14]: we you can reuse the function `import {findByTextContent} from 'sentry-test/utils'; `here

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 18]
```diff
@@ -674,355 +605,637 @@ describe('IssueList', function () {
...
+        });
 
-      expect(wrapper.find('Pagination Button').first().prop('disabled')).toBe(false);
+        await waitFor(() => {
+          expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+        });
 
-      // Click next again
-      wrapper.find('Pagination Button').last().simulate('click');
+        expect(await wrapper.findByText('Waiting for events…')).toBeInTheDocument();
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:981]: 👏👏👏👏

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 19]
```diff
@@ -674,355 +605,637 @@ describe('IssueList', function () {
...
-      };
-      expect(browserHistory.push).toHaveBeenLastCalledWith(pushArgs);
-      wrapper.setProps({location: pushArgs});
+        await waitFor(() => {
+          expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+        });
 
-      // Click previous
-      wrapper.find('Pagination Button').first().simulate('click');
+        expect(wrapper.queryByTestId('awaiting-events')).not.toBeInTheDocument();
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:1031]:
```suggestion
        expect(wrapper.queryByText('Waiting for events…')).toBe(null);
```

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 20]
```diff
@@ -674,355 +605,637 @@ describe('IssueList', function () {
...
+          organization: TestStubs.Organization({
+            projects,
+          }),
+        });
+
+        await waitFor(() => {
+          expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+        });
+
+        expect(wrapper.queryByTestId('awaiting-events')).not.toBeInTheDocument();
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:1139]:
```suggestion
  expect(wrapper.queryByText('Waiting for events…')).toBe(null);
```

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 21]
```diff
@@ -674,355 +605,637 @@ describe('IssueList', function () {
...
-          query: 'is:unresolved',
-          statsPeriod: '14d',
-        },
+        await waitFor(() => {
+          expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+        });
+
+        await waitFor(() =>
+          expect(wrapper.getByTestId('awaiting-events')).toBeInTheDocument()
+        );
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:1088]:
```suggestion
     expect(await wrapper.findByText('Waiting for events…')).toBeInTheDocument();
```

File: static/app/components/stream/processingIssueHint.tsx [hunk 22]
```diff
@@ -75,7 +75,7 @@ function ProcessingIssueHint({orgId, projectId, issue, showProject}: Props) {
 
   return (
     <StyledAlert type={alertType} icon={icon}>
-      <Wrapper>
+      <Wrapper data-test-id="processing-issue-hint">
```
Comment (priscilawebdev) [inline static/app/components/stream/processingIssueHint.tsx:78]: is this `data-test-id `being used?

File: static/app/components/smartSearchBar/index.tsx [hunk 23]
```diff
@@ -1347,7 +1347,13 @@ class SmartSearchBar extends React.Component<Props, State> {
               query
             )}
           </Highlight>
-          {useFormWrapper ? <form onSubmit={this.onSubmit}>{input}</form> : input}
+          {useFormWrapper ? (
+            <form data-test-id="search-form" onSubmit={this.onSubmit}>
```
Comment (priscilawebdev) [inline static/app/components/smartSearchBar/index.tsx:1351]: should we use `role='form'`instead of `data-test-id`  here?

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 24]
```diff
@@ -465,51 +421,54 @@ describe('IssueList', function () {
...
-      await tick();
-      wrapper.update();
 
-      wrapper.find('SavedSearchTab DropdownMenu a').simulate('click');
-      wrapper.find('SavedSearchMenuItem a').last().simulate('click');
+      await waitFor(() => {
+        expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+      });
+
+      fireEvent.click(wrapper.getByTestId('saved-search-title'));
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:438]:
```suggestion
      fireEvent.click(wrapper.getByText('Saved Searches'));
```

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 25]
```diff
@@ -605,62 +577,21 @@ describe('IssueList', function () {
...
-      await tick();
-      wrapper.update();
-
-      expect(browserHistory.push).toHaveBeenLastCalledWith(
-        expect.objectContaining({
-          pathname: '/organizations/org-slug/issues/searches/666/',
-          query: {},
-          search: '',
-        })
+      expect(wrapper.getByTestId('saved-search-title')).toHaveTextContent(
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:580]:
```suggestion
expect(wrapper.queryByText('Saved Searches')).toBe(null);
      expect(wrapper.getByText('Custom Search')).toBeInTheDocument(); 
```

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 26]
```diff
@@ -674,355 +605,637 @@ describe('IssueList', function () {
...
-
-      expect(getSavedSearchTitle(wrapper)).toBe('Unresolved TypeErrors');
+      await waitFor(() => {
+        expect(wrapper.queryByTestId('loading-indicator')).toBe(null);
+      });
 
-      // Select other saved search
-      wrapper.find('SavedSearchTab DropdownMenu a').simulate('click');
-      wrapper.find('SavedSearchMenuItem a').last().simulate('click');
+      fireEvent.click(wrapper.getByTestId('saved-search-title'));
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:642]:
```suggestion
    fireEvent.click(wrapper.getByText('Saved Searches'));
```

File: tests/js/spec/views/issueList/overview.spec.jsx [hunk 27]
```diff
@@ -674,355 +605,637 @@ describe('IssueList', function () {
...
 
-      wrapper.setProps({
-        params: {
-          ...router.params,
-          searchId: '234',
-        },
-      });
+      expect(wrapper.getByTestId('saved-search-title')).toHaveTextContent(
+        'Unresolved TypeErrors'
+      );
```
Comment (priscilawebdev) [inline tests/js/spec/views/issueList/overview.spec.jsx:659]:
```suggestion
expect(wrapper.queryByText('Saved Searches')).toBe(null);
      expect(wrapper.getByText('Unresolved TypeErrors')).toBeInTheDocument(); 
```

not sure if we will get these strings but we could only use Text queries here

File: tests/js/spec/views/issueList/overview.spec.jsx (hunk 20 above)
Comment (billyvg) [reply tests/js/spec/views/issueList/overview.spec.jsx:1139]: What happens if we change the `Waiting for events` text to something else? this will always pass

Comment (priscilawebdev) [reply tests/js/spec/views/issueList/overview.spec.jsx:1137]: yes .. I was not so sure about this one too, but at this point this text is no longer in the dom

**Review by scttcper — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Anything we can do with this?

**Comment by billyvg** [comment]
Closing this for now as I don't have time to get this fixed up rn
